apps/core/src/hooks/useGetKioskContents.ts

```diff
@@ -1,71 +1,102 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { SuiAddress, SuiObjectResponse } from '@mysten/sui.js';
+import { JsonRpcProvider, SuiAddress, SuiObjectResponse } from '@mysten/sui.js';
+import { fetchKiosk, getOwnedKiosks } from '@mysten/kiosk';
 import { useQuery } from '@tanstack/react-query';
 import { useRpcClient } from '../api/RpcClientContext';
-import { useGetOwnedObjects } from './useGetOwnedObjects';
+
+const getKioskId = (obj: SuiObjectResponse) =>
+	obj.data?.content &&
+	'fields' in obj.data.content &&
+	(obj.data.content.fields.for ?? obj.data.content.fields.kiosk);
 
 // OriginByte module for mainnet (we only support mainnet)
 export const ORIGINBYTE_KIOSK_MODULE =
 	'0x95a441d389b07437d00dd07e0b6f05f513d7659b13fd7c5d3923c7d9d847199b::ob_kiosk' as const;
 export const ORIGINBYTE_KIOSK_OWNER_TOKEN = `${ORIGINBYTE_KIOSK_MODULE}::OwnerToken`;
 
-const KIOSK_MODULE = '0x2::kiosk';
-const KIOSK_OWNER_CAP = `${KIOSK_MODULE}::KioskOwnerCap`;
+async function getOriginByteKioskContents(address: SuiAddress, rpc: JsonRpcProvider) {
+	const data = await rpc.getOwnedObjects({
+		owner: address,
+		filter: {
+			StructType: ORIGINBYTE_KIOSK_OWNER_TOKEN,
+		},
+	});
+	const ids = data.data.flatMap((object) => getKioskId(object) ?? []);
 
-const getKioskId = (obj: SuiObjectResponse) =>
-	obj.data?.content &&
-	'fields' in obj.data.content &&
-	(obj.data.content.fields.for ?? obj.data.content.fields.kiosk);
+	// fetch the user's kiosks
+	const ownedKiosks = await rpc.multiGetObjects({
+		ids: ids,
+		options: {
+			showContent: true,
+		},
+	});
 
-export function useGetKioskContents(address?: SuiAddress | null, disableOriginByteKiosk?: boolean) {
-	const rpc = useRpcClient();
-	const { data } = useGetOwnedObjects(address, {
-		MatchAny: [
-			{ StructType: KIOSK_OWNER_CAP },
-			...(!disableOriginByteKiosk ? [{ StructType: ORIGINBYTE_KIOSK_OWNER_TOKEN }] : []),
-		],
+	// find object IDs within a kiosk
+	const kioskObjectIds = await Promise.all(
+		ownedKiosks.map(async (kiosk) => {
+			if (!kiosk.data?.objectId) return [];
+			const objects = await rpc.getDynamicFields({
+				parentId: kiosk.data.objectId,
+			});
+			return objects.data.map((obj) => obj.objectId);
+		}),
+	);
+
+	// fetch the contents of the objects within a kiosk
+	const kioskContent = await rpc.multiGetObjects({
+		ids: kioskObjectIds.flat(),
+		options: {
+			showDisplay: true,
+			showType: true,
+		},
 	});
 
-	// find list of kiosk IDs owned by address
-	const kioskIds = data?.pages.flatMap((page) => page.data).map((obj) => getKioskId(obj)) ?? [];
+	return kioskContent;
+}
 
-	return useQuery({
-		queryKey: ['get-kiosk-contents', address, kioskIds],
-		queryFn: async () => {
-			if (!kioskIds.length) return [];
+async function getSuiKioskContents(address: SuiAddress, rpc: JsonRpcProvider) {
```

Comment:
will we be able to use the Kiosk SDK for any of this stuff in the future?

Reply:
I think it does that properly. Here, we're making sure we show contents from all owned kiosks of an address (because a user might have more than 1 kiosk), plus fetching the actual objects (kiosk returns an id).

One thing that's missing here is mapping the `kioskItem` data, to know which items are locked, listed etc.
But I think @mamos-mysten said he'll take care of it on a next PR. (You could have a look at Suifrens, I did that tweak there).

Reply:
Yea, I had that code added but removed for now since it was looking like a larger change w/ OB Kiosk support as well

Reply:
and yea, I'd like to move even more of this logic into the SDK where we can. I imagine it would be pretty common to need to fetch the kiosk items, and then immediately after call to `multiGetObjects` to retrieve the object info